openedx/features/calendar_sync/ics.py

```diff
@@ -46,28 +48,31 @@ def generate_ics_for_event(uid, title, course_name, now, start, organizer_name,
     return cal.to_ical()
 
 
-def generate_ics_for_user_course(course, user, request):
+def generate_ics_files_for_user_course(course, user, user_calendar_sync_config_instance):
     """
     Generates ics-formatted bytestrings of all assignments for a given course and user.
 
     To pretty-print each bytestring, do: `ics.decode('utf8').replace('\r\n', '\n')`
 
-    Returns an iterable of ics files, each one representing an assignment.
+    Returns a dictionary of ics files, each one representing an assignment.
     """
-    assignments = get_course_assignments(course.id, user, request)
+    assignments = get_course_assignments(course.id, user)
     platform_name = get_value('platform_name', settings.PLATFORM_NAME)
     platform_email = get_value('email_from_address', settings.DEFAULT_FROM_EMAIL)
     now = datetime.now(pytz.utc)
+    site_config = SiteConfiguration.get_configuration_for_org(course.org)
 
-    return (
-        generate_ics_for_event(
+    ics_files = {}
+    for assignment in assignments:
+        ics_files[assignment.title] = generate_ics_for_event(
```

Comment:
You *could* do a dictionary comprehension here with `{ assignment.title: generate_ics_for_event(...` but this is fine too.

Reply:
Ya the ics file needs to have a sequence property in order to validate that the previous event is to be edited.

Reply:
MT loves dictionary comprehensions. Potentially more than list comprehensions.